sdk/framework/backend.go

```diff
@@ -225,6 +226,16 @@ func (b *Backend) HandleRequest(ctx context.Context, req *logical.Request) (*log
 
 	if path.Operations != nil {
 		if op, ok := path.Operations[req.Operation]; ok {
+
+			// Check whether this operation should be forwarded
+			replState := b.System().ReplicationState()
+			props := op.Properties()
+
+			if !b.System().LocalMount() && (props.ForwardPerformanceStandby && replState.HasState(consts.ReplicationPerformanceStandby) ||
```

Comment:
This may be clearer as two different if statements: 
```
if props.ForwardPerformanceStandby && replState.HasState(consts.ReplicationPerformanceStandby) {
	return nil, logical.ErrReadOnly
}

if props.ForwardPerformanceSecondary && !b.System().LocalMount() && replState.HasState(consts.ReplicationPerformanceSecondary)) {
	return nil, logical.ErrReadOnly
}
```

Reply:
Done